target/cheri-common/cheri-lazy-capregs.h

```diff
@@ -236,6 +236,33 @@ static inline void update_capreg(CPUArchState *env, unsigned regnum,
     cheri_log_instr_changed_gp_capreg(env, regnum, target);
 }
 
+static inline void update_capreg_cursor_from(CPUArchState *env, unsigned regnum,
+                                             const cap_register_t *source_cap,
+                                             unsigned source_regnum,
+                                             const target_ulong new_cursor)
+{
+    /*
+     * This function can be called to avoid copying a full cap_register_t when
+     * updating a capability register inplace.
```

Comment:
Maybe move this above the function?  It's also probably good to give some more motivation, and also say something like "`new_cursor` must be representable given the bounds of `source_cap`; see `addr_in_cap_bounds` and `is_representable_cap_with_addr`."